### PR TITLE
chore: log internal errors with more details

### PR DESCRIPTION
**Pull request by miparnisari**
Log all errors, and especially log internal server errors.

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -137,11 +137,11 @@ func New(dependencies *Dependencies, config *Config) (*Server, error) {
 		defaultServeMuxOpts: []runtime.ServeMuxOption{
 			runtime.WithForwardResponseOption(httpmiddleware.HTTPResponseModifier),
 			runtime.WithErrorHandler(func(c context.Context, sr *runtime.ServeMux, mm runtime.Marshaler, w http.ResponseWriter, r *http.Request, e error) {
-				actualCode := serverErrors.ConvertToEncodedErrorCode(status.Convert(e))
-				if serverErrors.IsValidEncodedError(actualCode) {
-					dependencies.Logger.ErrorWithContext(c, "grpc error", logger.Error(e), logger.String("request_url", r.URL.String()))
+				dependencies.Logger.ErrorWithContext(c, "grpc error", logger.Error(e), logger.String("request_url", r.URL.String()))
```
Comment (craigpastro) [inline server/server.go:140]: Would this make filtering a bit nicer?
```suggestion
				dependencies.Logger.ErrorWithContext(c, "grpc_error", logger.Error(e), logger.String("request_url", r.URL.String()))
```

File: server/server.go [hunk 2]
```diff
@@ -137,11 +137,11 @@ func New(dependencies *Dependencies, config *Config) (*Server, error) {
...
+				if internalError, ok := e.(serverErrors.InternalError); ok {
+					dependencies.Logger.ErrorWithContext(c, "internal error", logger.Error(internalError.Internal()))
```
Comment (craigpastro) [inline server/server.go:142]: Similarly here?
```suggestion
					dependencies.Logger.ErrorWithContext(c, "internal_error", logger.Error(internalError.Internal()))
```
Or should this also be a `grpc_error` so we can filter all errors and have an extra field like `"error_type": "internal"` or something?

Comment (elbuo8) [reply server/server.go:142]: I'm actually wondering if we can perform a single log. Building a `fields` array that gets extended if it's an `InternalError` and with a single log you can see everything.

**Review by elbuo8 — APPROVED** [review]

**Review by craigpastro — APPROVED** [review]
LGTM

**Pull request merged**